File: fix_id.js
const fs = require('fs');
let code = fs.readFileSync('build_ui_premium_v3.js', 'utf8');

const t = 'font-family:var(--font-arabic, \\'Amiri\\'); font-size:36px; line-height:2; text-align:right; margin-bottom:20px; color:var(--accent-gold);">${a.teksArab}</div>';
const target = '<div class="notranslate" style="' + t;
const replacement = '<div id="text-arab-${a.nomorAyat}" class="notranslate" style="' + t;

code = code.replace(target, replacement);

fs.writeFileSync('build_ui_premium_v3.js', code);
console.log('Fixed Tajwid ID!');

File: fix_id.py
import sys

with open('build_ui_premium_v3.js', 'r', encoding='utf-8') as f:
    lines = f.read().split('\\n')

for i in range(len(lines)):
    if 'class="notranslate"' in lines[i] and 'style="font-family:var(--font-arabic' in lines[i] and '${a.teksArab}' in lines[i]:
        # We want the file to have exactly: id="text-arab-${a.nomorAyat}"
        lines[i] = lines[i].replace('<div class="notranslate"', '<div id="text-arab-${a.nomorAyat}" class="notranslate"')

with open('build_ui_premium_v3.js', 'w', encoding='utf-8') as f:
    f.write('\\n'.join(lines))

print("Fixed with Python!")
